main.go

```diff
@@ -1,137 +1,78 @@
 package main
 
 import (
-	"flag"
+	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
 	"os/exec"
 	"path"
 	"strings"
 
-	"github.com/shoukoo/terra-map/service"
 	"gopkg.in/yaml.v2"
 )
 
 var dir string
-var validResource = []string{"aws_instance", "aws_s3_bucket", "aws_sqs_queue"}
 
-func init() {
-	flag.StringVar(&dir, "d", "", "specify a dir path to generate a map.yml")
-	flag.Parse()
+type Resource interface {
+	Process(resource []string, b []byte) []interface{}
+	Conditions() []Condition
 }
```

Comment:
If you decide to add more command line arguments, then yes we should add the flag library back in. 
Otherwise I'd aim to keep things as minimalistic as possible.